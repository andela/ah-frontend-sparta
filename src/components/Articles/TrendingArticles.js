import React from 'react';
import moment from 'moment';
import BookmarkImage from '../../assets/images/bookmark.png';

const getRandomArticles = (count, articlesArray) => {
	// Make a copy of the array
	const temporaryArray = articlesArray.slice(articlesArray);
	const returned_array = [];

	for (var i = 0; i < count; i++) {
		var index = Math.floor(Math.random() * temporaryArray.length);
		var removed = temporaryArray.splice(index, 1);
		// Since we are only removing one element
		returned_array.push(removed[0]);
	}
	return returned_array;
};

const TrendingArticles = (props) => {
	const { articles } = props;
	const randomArray = getRandomArticles(5, articles);
	return (
		<div>
			<h1>Trending</h1>
			<br />
			<hr />
			<br />
			{articles.length > 0 ? (
				randomArray.map((article) => {
					let article_date = moment(article.createdAt.slice(0, 10)).format('LL');
					return (
						<div className='trending' key={article.id}>
							<div className='trending-articles'>
								<h4>{article.title}</h4>
								<p className='author_name'>
									Created By <b>{article.author.username}</b>
								</p>
								<h6>
									{article_date} - {article.article_read_time}
								</h6>
							</div>
							<div className='book-mark-holder'>
								<img src={BookmarkImage} className='bookmark-image' />
							</div>
							<br />
						</div>
					);
				})
			) : (
				<div>No Trending Articles where Found !!!!</div>
			)}
		</div>
	);
};

export default TrendingArticles;
