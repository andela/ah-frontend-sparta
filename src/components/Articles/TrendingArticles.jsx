import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import DisplayTrendingArticles from './DisplayTrendingArticles';
import getRandomArticles from '../../helpers/randomArticles';

const TrendingArticles = ({ articles }) => {
  const randomArray = getRandomArticles(5, articles);
  return (
    <div>
      <h1>Trending</h1>
      <br />
      <hr />
      <br />
      {articles.length > 0 ? (
        randomArray.map((article) => {
          const articleDate = moment(
            article.createdAt.slice(0, 10),
          ).format('LL');
          return (
            <div key={article.id}>
              <DisplayTrendingArticles
                articleID={article.id}
                trendingArticles={article}
                articleDate={articleDate}
              />
            </div>
          );
        })
      ) : (
        <div>No Trending Articles where Found !!!!</div>
      )}
    </div>
  );
};

TrendingArticles.defaultProps = {
  articles: [],
};
TrendingArticles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.object),
};

export default TrendingArticles;
