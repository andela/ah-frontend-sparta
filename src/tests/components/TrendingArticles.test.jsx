import React from 'react';
import { mount } from 'enzyme';
import TrendingArticles from '../../components/Articles/TrendingArticles';


describe('Articles Component', () => {
  let mountApp;
  const props = {
    articles: [

      {
        id: 1,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },
      {
        id: 2,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },
      {
        id: 3,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },
      {
        id: 4,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },
      {
        id: 5,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },
      {
        id: 6,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },
      {
        id: 7,
        author: {
          username: 'rogha',
          firstname: null,
          lastname: null,
          bio: null,
          image: null,
          followers_no: 1,
          following_no: 0,
        },
        average_rating: null,
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
        updatedAt: '2019-04-18T11:55:58.111902Z',
        tags: [],
        likes: 0,
        dislikes: 0,
      },

    ],

  };
  beforeEach(() => {
    mountApp = mount(<TrendingArticles {...props} />);
  });

  it('Should render without crashing', () => {
    expect(mountApp).toMatchSnapshot();
  });

  it('Should render articles', () => {
    expect(mountApp.find('.trending-articles')).toHaveLength(5);
  });
});
