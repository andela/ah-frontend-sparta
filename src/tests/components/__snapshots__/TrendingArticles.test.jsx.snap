// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Component Should render without crashing 1`] = `ReactWrapper {}`;
